src/index.ts

```diff
@@ -330,45 +330,46 @@ try {
   // fetch all existing loans (or up to the configured limit)
   if (!existingSubscription || !status.synced) {
     // The API can provide the total loan count, but is currently capped at 1000
-    const loanCount = await getFeedTotalCount(apiClient, "LoansFeed");
+    const loanCount = await getTotalCount(apiClient, "loans");
     const maxLoanCount = 1000;
-    const loanCountDisplay =
-      loanCount > maxLoanCount ? `>${maxLoanCount}` : String(loanCount);
 
-    const loansFeedQuery = makeFeedQuery(
-      "LoansFeed",
-      selectSubfields("loans", loanSelectionSet)
-    );
-    const limit = 10;
+    const loansQuery = makeQuery("loans", loanSelectionSet.join(" "));
+
+    const first = 10;
     const { loanImportLimit } = configuration;
     logger.info(
       `Performing initial import of ${loanImportLimit ?? "all"} loans`
     );
-    for (let offset = 0; ; offset += limit) {
-      const endIndex =
-        loanCount <= maxLoanCount
-          ? Math.min(offset + limit, loanCount)
-          : offset + limit;
-      logger.info(
-        `Fetching loans ${offset + 1}-${endIndex} of ${loanCountDisplay}`
+    let after: string | null = null;
+    let hasNextPage = true;
+    let importedLoansCount = 0;
+    while (hasNextPage) {
+      const data: LoansOutput = await getData<LoansOutput>(
+        apiClient,
+        loansQuery,
+        {
+          first,
+          after,
+          orderBy: [{ fieldKey: "createdAt", sortOrder: "asc" }],
+        }
       );
-      const output = await getFeed<LoansFeedOutput>(apiClient, loansFeedQuery, {
-        limit,
-        offset,
-        orderBy: [{ fieldKey: "createdAt", sortOrder: "asc" }],
-      });
-      const { loans } = output.LoansFeed;
+      const {
+        loans: { edges, pageInfo },
+      } = data;
+      const loans = edges.map(({ node }) => node);
+      after = pageInfo.endCursor;
+      hasNextPage = pageInfo.hasNextPage;
       for (const loan of loans) {
         await loanDatabase.write(loan.id, loan);
       }
-      const imported = offset + loans.length;
+      importedLoansCount = importedLoansCount + loans.length;
```

Comment:
Slight readability improvement:

```suggestion
      importedLoansCount += loans.length;
```